pgo/cmd/clone.go

```diff
@@ -0,0 +1,100 @@
+// Package cmd provides the command line functions of the crunchy CLI
+package cmd
+
+/*
+ Copyright 2019 Crunchy Data Solutions, Inc.
+ Licensed under the Apache License, Version 2.0 (the "License");
+ you may not use this file except in compliance with the License.
+ You may obtain a copy of the License at
+
+      http://www.apache.org/licenses/LICENSE-2.0
+
+ Unless required by applicable law or agreed to in writing, software
+ distributed under the License is distributed on an "AS IS" BASIS,
+ WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ See the License for the specific language governing permissions and
+ limitations under the License.
+*/
+
+import (
+	"fmt"
+	"os"
+
+	msgs "github.com/crunchydata/postgres-operator/apiservermsgs"
+	"github.com/crunchydata/postgres-operator/pgo/api"
+	log "github.com/sirupsen/logrus"
+	"github.com/spf13/cobra"
+)
+
+var (
+	// the source cluster used for the clone, e.g. "oldcluster"
+	SourceClusterName string
+	// the taregt/destination cluster used for the clone, e.g. "newcluster"
+	TargetClusterName string
+)
+
+var cloneCmd = &cobra.Command{
+	Use:   "clone",
+	Short: "Copies the primary database of an existing cluster to a new cluster",
+	Long: `Clone makes a copy of an existing PostgreSQL cluster managed by the Operator and creates a new PostgreSQL cluster managed by the Operator, with the data from the old cluster.
+
+	pgo clone oldcluster newcluster`,
+	Run: func(cmd *cobra.Command, args []string) {
+		// if the namespace is not specified, default to the PGONamespace specified
+		// in the `PGO_NAMESPACE` environmental variable
+		if Namespace == "" {
+			Namespace = PGONamespace
```

Comment:
Just food for thought, what about defaulting to the kubeconfig default namespace? I haven't verified, but seems like that's what the apiserver _should_ do in the case of receiving a request with an empty namespace.

Reply:
@abrightwell Not a bad idea, but that would be a separate patch as that would effectively be a global change. Please feel free to propose it :wink: